### ISSUE TITLE
"Create test" action mangles text when there are unsaved changes in buffer of test file (emacs cider)

### ISSUE THREAD

**Issue by bo-tato** [clojure-lsp/clojure-lsp#1458]
**Describe the bug**
"Create test" 

**To Reproduce**
1. Make a scratch project `neil new scratch foo`
2. open `src/foo/foo.clj` and with cursor on (defn exec... do "create test" code action
3. creates file and test perfect, see first screenshot
4. but if we look on disk, test/foo/foo_test.clj is created but empty, in the emacs buffer the changes are unsaved
5. back in foo.clj do create test again, see 2nd screenshot test is inserted in middle of namespace heading
6. undo that step 5, and save the

 foo_test.clj file to disk as it was after step 4

**Expected behavior**

**Screenshots**
![2023-01-20-170546_736x648_scrot](https://user-images.githubusercontent.com/122528427/213747103-774d3c01-b8ca-4254-b894-6cd39365d687.png)
![2023-01-20-170910_736x648_scrot](https://user-images.githubusercontent.com/122528427/213747593-8a67c4eb-4116-4cb0-8ec0-8f0811b42c2b.png)

<details>
 <summary><b>Log - client <-> server</b></summary>
<pre>
ADD JSON HERE
</pre>
</details>

<details>
 <summary><b>Log - clojure-lsp</b></summary>
<pre>
ADD HERE
</pre>
</details>

**User details (please complete the following information):**
 - OS: [e.g. ArchLinux, MacOS, Windows 10]
 - Editor [e.g. emacs, nvim, VSCode (Calva)]
 - Version: (post the result of `clojure-lsp --version`)

**Additional context**

**Comment by ericdallo** [comment]
I'm not sure if this is a issue on server side or lsp-mode, will take a closer look this weekend, one way to confirm that is to check client<->server logs

**Comment by bo-tato** [comment]
https://gist.github.com/bo-tato/3f79ea9d7e4269b99efa52de5028529b
I think this has logs of all request and response between emacs and clojure-lsp, is that what you mean or some other logs?

**Comment by ericdallo** [comment]
yes, maybe there lots of noise there we could reduce, but we should look for applyEdit server response and check if makes sense
